### PR TITLE
Remove v1alpha3 from v1

### PR DESCRIPTION
**Pull request by ijsnellf**
V2 should soon have more complete v1alpha3 functionality than v1. Once we have tests in place proving this we should remove the old v1 implementation.  This PR does that.

### CODE REVIEW

File: pilot/pkg/proxy/envoy/v1/mock/discovery.go [hunk 1]
```diff
@@ -62,6 +62,7 @@ var (
 		ID:        "ingress.default",
 		Domain:    "default.svc.cluster.local",
 	}
+	// TODO: remove?
```
Comment (rshriram) [inline pilot/pkg/proxy/envoy/v1/mock/discovery.go]: yes

File: pilot/pkg/proxy/envoy/v1/mock/discovery.go [hunk 2]
```diff
@@ -115,6 +116,7 @@ func MakeService(hostname, address string) *model.Service {
 }
 
 // MakeExternalHTTPService creates mock external service
+// TODO: remove?
```
Comment (rshriram) [inline pilot/pkg/proxy/envoy/v1/mock/discovery.go]: no

Comment (ijsnellf) [reply pilot/pkg/proxy/envoy/v1/mock/discovery.go]: nvm, this is k8s external services --not v1alpha3 external services. it should stay.

**Pull request merged**